src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/SourceInfo.cs

```diff
@@ -40,17 +41,25 @@ public SourceInfo(string fullTypeName, bool isInterface, ImmutableHashSet<string
         /// </summary>
         public ImmutableHashSet<string> TaintedProperties { get; }
 
+        public delegate bool ArgumentCheck(ImmutableHashSet<object> argumentValue);
+
         /// <summary>
         /// Methods that generate tainted data.
         /// </summary>
-        public ImmutableHashSet<string> TaintedMethods { get; }
+        public ImmutableDictionary<string, ImmutableDictionary<string, ArgumentCheck>> TaintedMethods { get; }
```

Comment:
With more complicated data structures like dictionaries of dictionaries, I like to comment how the mapping works (Method name -> Argument name -> ArgumentCheck predicate).